I'm not sure what's going on with the pylint check failing -- looking at the logs, it appears that setting up the test environment is failing, which I'm pretty sure has nothing to do with my code.  

Looks like the failure in azure pipelines is [here](https://dev.azure.com/home-assistant/Home%20Assistant/_build/results?buildId=6657&view=logs&j=8b66295a-05e1-52cf-5640-7a4f4e1975e2&t=e54c6b36-aecc-5497-42ac-744c0780ce56&l=220):

    Collecting bimmer_connected==0.5.6 (from -r requirements_all.txt (line 269))  
      ERROR: Could not find a version that satisfies the requirement bimmer_connected==0.5.6 (from -r 
    requirements_all.txt (line 269)) (from versions: 0.0.1, 0.0.3, 0.0.4, 0.1.0, 0.2.0, 0.3.0, 0.4.0.dev0, 0.4.0, 0.4.1, 0.5.0, 0.5.1, 0.5.2, 0.5.3, 0.5.9b0, 0.5.9b1)
    ERROR: No matching distribution found for bimmer_connected==0.5.6 (from -r requirements_all.txt (line 269))
    ##[error]Bash exited with code '1'.
    Finishing: Create Virtual Environment & Install Requirements

--- a/homeassistant/components/sonos/media_player.py
+++ b/homeassistant/components/sonos/media_player.py
@@ -15,6 +15,7 @@ from homeassistant.components.media_player import MediaPlayerDevice
 from homeassistant.components.media_player.const import (
     ATTR_MEDIA_ENQUEUE,
     MEDIA_TYPE_MUSIC,
+    MEDIA_TYPE_PLAYLIST,
     SUPPORT_CLEAR_PLAYLIST,
     SUPPORT_NEXT_TRACK,
     SUPPORT_PAUSE,
@@ -943,7 +944,7 @@ class SonosEntity(MediaPlayerDevice):
 
         If ATTR_MEDIA_ENQUEUE is True, add `media_id` to the queue.
         """
-        if media_type == "playlist":
+        if media_type == MEDIA_TYPE_PLAYLIST:
             try:
                 playlists = self.soco.get_sonos_playlists()
                 playlist = next(p for p in playlists if p.title == media_id)

--- a/homeassistant/components/sonos/media_player.py
+++ b/homeassistant/components/sonos/media_player.py
@@ -950,9 +950,12 @@ class SonosEntity(MediaPlayerDevice):
                 self.soco.clear_queue()
                 self.soco.add_to_queue(playlist)
                 self.soco.play_from_queue(0)
+                return
             except StopIteration:
                 _LOGGER.error('Could not find a Sonos playlist named "%s".', media_id)
-        elif kwargs.get(ATTR_MEDIA_ENQUEUE):
+                # Fall through to the previous behavior, in case someone is relying on
+                # passing a URI with a media_type of "playlist"
+        if kwargs.get(ATTR_MEDIA_ENQUEUE):
             try:
                 self.soco.add_uri_to_queue(media_id)
             except SoCoUPnPException:

--- a/homeassistant/components/sonos/media_player.py
+++ b/homeassistant/components/sonos/media_player.py
@@ -944,30 +944,30 @@ class SonosEntity(MediaPlayerDevice):
 
         If ATTR_MEDIA_ENQUEUE is True, add `media_id` to the queue.
         """
-        if media_type == MEDIA_TYPE_PLAYLIST:
+        if media_type == MEDIA_TYPE_MUSIC:
+            if kwargs.get(ATTR_MEDIA_ENQUEUE):
+                try:
+                    self.soco.add_uri_to_queue(media_id)
+                except SoCoUPnPException:
+                    _LOGGER.error(
+                        'Error parsing media uri "%s", '
+                        "please check it's a valid media resource "
+                        "supported by Sonos",
+                        media_id,
+                    )
+            else:
+                self.soco.play_uri(media_id)
+        elif media_type == MEDIA_TYPE_PLAYLIST:
             try:
                 playlists = self.soco.get_sonos_playlists()
                 playlist = next(p for p in playlists if p.title == media_id)
                 self.soco.clear_queue()
                 self.soco.add_to_queue(playlist)
                 self.soco.play_from_queue(0)
-                return
             except StopIteration:
-                _LOGGER.error('Could not find a Sonos playlist named "%s".', media_id)
-                # Fall through to the previous behavior, in case someone is relying on
-                # passing a URI with a media_type of "playlist"
-        if kwargs.get(ATTR_MEDIA_ENQUEUE):
-            try:
-                self.soco.add_uri_to_queue(media_id)
-            except SoCoUPnPException:
-                _LOGGER.error(
-                    'Error parsing media uri "%s", '
-                    "please check it's a valid media resource "
-                    "supported by Sonos",
-                    media_id,
-                )
+                _LOGGER.error('Could not find a Sonos playlist named "%s"', media_id)
         else:
-            self.soco.play_uri(media_id)
+            _LOGGER.error('Sonos does not support a media type of "%s"', media_type)
 
     @soco_error()
     def join(self, slaves):

